### PR TITLE
remove-singleton: replace SwapApp with this.app and pass from constructor

### PR DESCRIPTION
**Pull request by caffeinum**
## Description

This removes singleton to facilitate easy testing etc.

## What to test

```
cd examples/react
npm i
npm run start
```

```
npx jest tests/
```

```
cd simple
npm i
npm run test
```

### CODE REVIEW

File: tests/ETHTOKEN2USDT.test.js [hunk 1]
```diff
@@ -10,6 +10,8 @@ jest.mock('swap.app')
 jest.unmock('swap.flows')
 jest.unmock('swap.swaps')
 
+let app = SwapApp
```
Comment (shendel) [inline tests/ETHTOKEN2USDT.test.js]: тык

File: src/swap.orders/Order.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Order {
    * @param {number}  data.buyAmount
    * @param {number}  data.sellAmount
    */
-  constructor(parent, data) {
+  constructor(app, parentCollection, data) {
```
Comment (shendel) [inline src/swap.orders/Order.js]: тык #2

**Comment by caffeinum** [comment]
### Breaking changes

#### Public

##### SwapApp

first setup app:

```javascript
import SwapApp from 'swap.app'

SwapApp.setup(...)
```
then use:

```javascript
// before:
import SwapApp from 'swap.app'

SwapApp.services.orders.getMyOrders()

// after:
import SwapApp from 'swap.app'

SwapApp.shared().services.orders.getMyOrders()
```

##### Swap

When you create `Swap`, you need to pass your currently used app instance! This is a change in public interface!

```javascript
before:
- const swap = new Swap(orderId)
after:
+ const swap = new Swap(orderId, app)
```

##### Flow
Everywhere inside Flow and Swap code, you now should use `this.app` to access what previously was `SwapApp`, for example, `SwapApp.services.auth` or `SwapApp.services.orders`

```javascript
- const balance = await this.btcSwap.fetchBalance(SwapApp.services.auth.accounts.btc.getAddress())
+ const balance = await this.btcSwap.fetchBalance(this.app.services.auth.accounts.btc.getAddress())
```

##### Flow.read, Swap.read

These are recently introduced methods. They are static and are to read swap data without constructing an object and launching a Swap.

```javascript
- Flow.read({ id: "QmZZZ-123" })
+ Flow.read(app, { id: "QmZZZ-123" })
```


#### Private

##### ServiceInterface

When connected to SwapApp, it is automatically to this.app of service instance.

There are no changes in public scope.


```javascript
class ServiceInterface {
  ...
  _attachSwapApp(app) {
    SwapApp.required(app)

    this.app = app
  }
  ...
}
```

##### SwapInterface

BtcSwap, EthSwap etc connect to current SwapApp on init. This happens automatically.


```javascript
class SwapInterface {
   _initSwap(app) {}
}

class BtcSwap extends SwapInterface {
   ...
   _initSwap(app) {
    this.app = app
     ....
   }
}

```

##### Util - Actual Key

A bit changed call style:

```javascript
- create(keyName)
+ create(SwapApp.shared(), keyName)

- compare(keyName, actualKey)
+ compare(SwapApp.shared(), keyName, actualKey)

- remove(keyName)
+ remove(SwapApp.shared(), keyName)
```

##### SwapAuth modules

Account init functions now need app. Again, no changes in the public scope


```javascript
// btc.js

- login(_privateKey) { ... }
+ login(_privateKey, app) { ... }
```

##### Orders

Order was never created directly, but worth mentioning: the constructor has been changed.

```javascript
// SwapOrders.create():

const order = new Order(this, {
  id: id || getUniqueId(),
  ...
})
```

became

```javascript
const order = new Order(this.app, this, {
  id: id || this.getUniqueId(),
  ...
})
```

##### SwapRoom
This is the small module that is stored inside Swap object to communicate directly with participant.

Its' constructor changed, too:
```javascript
new SwapRoom(app, { swapId, participantPeer })
```

File: src/swap.swap/Swap.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import Room from './Room'
 
 class Swap {
 
-  constructor(id, order) {
+  constructor(id, app, order) {
```
Comment (shendel) [inline src/swap.swap/Swap.js]: app последним параметром

**Pull request merged**